docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+      --set fullnameOverride=postgresql \
+      --set postgresqlDatabase=products \
+      --set resources.requests.cpu=50m \
+      --set resources.requests.memory=156Mi \
+      --set readinessProbe.initialDelaySeconds=60 \
+      --set livenessProbe.initialDelaySeconds=60;`,
+    ];
+    await postgresql.run();
+  }
+};
+
+const provisionEnvironment = async (environmentName, projects) => {
+  await deployDependencies(environmentName);
+};
+
+```
+
+Before deploying our PostgreSQL StatefulSet we check if one already exists. 
+To list StatefulSets in a Namespace we need to use `listNamespacedStatefulSet` 
+api endpoint that lives in `kubernetes.Apps_v1Api` set of APIs. 
+We create instance of it called `k8sAppClient` first.
+
+Once we verify our dependency needs to be installed we create `postgresql` Job that will 
+run `lachlanevenson/k8s-helm:v2.12.3` image as it's worker. 
+Lachlan Evenson has been building and publishing [docker images](https://hub.docker.com/r/lachlanevenson/k8s-helm/) with every version of Helm.
+
+After Job successfully completes we can verify our PostgreSQL is installed:
+
+```console
+$ helm list
+
+NAME            	STATUS  	CHART                	NAMESPACE
+bob-postgresql  	DEPLOYED	postgresql-3.16.     	bob
+```
+
+## Environment ConfigMap
+
+Environment ConfigMap is created in `brigade` namespace where we keep track of all projects we would like to be installed in our environment. One ConfigMap is created per environment.
+
+Sample of data structure:
+
+```yaml
+projects:
+  products:
+    tag: prod
+  orders:
+    tag: prod
+```
+
+Our ConfigMap will be labeled with `type: preview-environment-config` label. This label will be used as a selector by application's `brigade.js` script at the time of a release.
+
+```js
+const createEnvironmentConfigMap = async (name, projects) => {
+  const configMap = new kubernetes.V1ConfigMap();
+  const metadata = new kubernetes.V1ObjectMeta();
+  metadata.name = `preview-environment-${name}`;
+  metadata.namespace = 'brigade';
+  metadata.labels = {
+    type: "preview-environment-config",
+    environmentName: name,
+  };
+  configMap.metadata = metadata;
+  configMap.data = {
+    projects: yaml.dump(projects),
+  };
+
+  await k8sCoreClient.createNamespacedConfigMap('brigade', configMap);
+};
+
+const provisionEnvironment = async (environmentName, projects) => {
+  await createEnvironmentConfigMap(environmentName, projects);
+};
+
+```
+
+You can see final implementation of the script in the reference repository:
+
+https://github.com/brigadecore/brigade-tutorial-config/blob/master/brigade.js
+
+## Products Project
+
+`Products Service` is a sample application that will be deployed to our preview environment every time it is released (tagged with a `prod` tag). Instance of PostgreSQL deployed  during creation of the environment is a dependency of our service.
+
+![Create Environment Pipeline](https://docs.brigade.sh/img/preview-environments-release.png)
```

Comment:
This picture doesn't seem to be displaying... at least in the docs preview:
![Screen Shot 2019-06-03 at 11 44 33 AM](https://user-images.githubusercontent.com/9322017/58822539-00e4ff00-85f5-11e9-88a5-7d595ade0562.png)

Reply:
Same as above...